# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/mediasink/MediaSink.h
# Commit: 1c45b135318d
# Full Hash: 1c45b135318d8cb519b8d58af72ab864fe6bfd62
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-11-14 17:12:31
# Regressor Bug: 1172394
# File Overlap Count: 1
# Description:
#   Bug 1172394 - Simplify MediaSink somewhat. r=padenot
#   
#   This patches does several minor things:
#   - Moves SetSink (from setSinkid) to automatic coalescing of multiple calls
#     through a Canonical/Mirror setup instead of a manual atomic counter.
# ==============================================================================

diff -r c57c41e8c39e -r 1c45b135318d dom/media/mediasink/MediaSink.h
--- a/dom/media/mediasink/MediaSink.h	Wed Nov 13 08:55:39 2019 +0000
+++ b/dom/media/mediasink/MediaSink.h	Wed Nov 13 08:55:54 2019 +0000
@@ -7,7 +7,6 @@
 #ifndef MediaSink_h_
 #define MediaSink_h_
 
-#include "AudioDeviceInfo.h"
 #include "MediaInfo.h"
 #include "mozilla/MozPromise.h"
 #include "mozilla/RefPtr.h"
@@ -39,23 +38,6 @@
   NS_INLINE_DECL_THREADSAFE_REFCOUNTING(MediaSink);
   typedef mozilla::TrackInfo::TrackType TrackType;
 
-  struct PlaybackParams {
-    PlaybackParams()
-        : mVolume(1.0), mPlaybackRate(1.0), mPreservesPitch(true) {}
-    double mVolume;
-    double mPlaybackRate;
-    bool mPreservesPitch;
-    RefPtr<AudioDeviceInfo> mSink;
-  };
-
-  // Return the playback parameters of this sink.
-  // Can be called in any state.
-  virtual const PlaybackParams& GetPlaybackParams() const = 0;
-
-  // Set the playback parameters of this sink.
-  // Can be called in any state.
-  virtual void SetPlaybackParams(const PlaybackParams& aParams) = 0;
-
   // EndedPromise needs to be a non-exclusive promise as it is shared between
   // both the AudioSink and VideoSink.
   typedef MozPromise<bool, nsresult, /* IsExclusive = */ false> EndedPromise;
@@ -100,6 +82,10 @@
   // Pause/resume the playback. Only work after playback starts.
   virtual void SetPlaying(bool aPlaying) = 0;
 
+  // Get the playback rate.
+  // Can be called in any state.
+  virtual double PlaybackRate() const = 0;
+
   // Single frame rendering operation may need to be done before playback
   // started (1st frame) or right after seek completed or playback stopped.
   // Do nothing if this sink has no video track. Can be called in any state.
@@ -122,6 +108,10 @@
   // Can be called in any state.
   virtual bool IsPlaying() const = 0;
 
+  // The audio output device this MediaSink is playing audio data to. The
+  // default device is used if this returns null.
+  virtual const AudioDeviceInfo* AudioDevice() { return nullptr; }
+
   // Called on the state machine thread to shut down the sink. All resources
   // allocated by this sink should be released.
   // Must be called after playback stopped.